src/Avalonia.Diagnostics/Diagnostics/Views/MainWindow.xaml.cs

```diff
@@ -99,34 +162,72 @@ private void RawKeyDown(RawKeyEventArgs e)
                 return;
             }
 
-            const RawInputModifiers modifiers = RawInputModifiers.Control | RawInputModifiers.Shift;
-
-            if (e.Modifiers == modifiers)
+            switch (e.Modifiers)
             {
-#pragma warning disable CS0618 // Type or member is obsolete
-                var point = (Root as IInputRoot)?.MouseDevice?.GetPosition(Root) ?? default;
-#pragma warning restore CS0618 // Type or member is obsolete                
+                case RawInputModifiers.Control | RawInputModifiers.Shift:
+                {
+                    IControl? control = null;
+
+                    foreach (var popupRoot in GetPopupRoots(Root))
+                    {
+                        control = GetHoveredControl(popupRoot);
+
+                        if (control != null)
+                        {
+                            break;
+                        }
+                    }
+
+                    control ??= GetHoveredControl(Root);
 
-                var control = Root.GetVisualsAt(point, x =>
+                    if (control != null)
                     {
-                        if (x is AdornerLayer || !x.IsVisible) return false;
-                        if (!(x is IInputElement ie)) return true;
-                        return ie.IsHitTestVisible;
-                    })
-                    .FirstOrDefault();
+                        vm.SelectControl(control);
+                    }
+
+                    break;
+                }
 
-                if (control != null)
+                case RawInputModifiers.Control | RawInputModifiers.Alt when e.Key == Key.F:
                 {
-                    vm.SelectControl((IControl)control);
+                    vm.FreezePopups = !vm.FreezePopups;
```

Comment:
This will toggle on and off instantly since this function receives both down and up events (name is wrong, you are receiving all kinds of key events here).